File: aat-android/src/main/java/ch/bailu/aat/views/preferences/SolidCheckListDialog.java
package ch.bailu.aat.views.preferences;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.DialogInterface.OnMultiChoiceClickListener;

import ch.bailu.aat.preferences.SolidCheckList;

public class SolidCheckListDialog extends AbsSolidDialog implements  OnMultiChoiceClickListener{
    private final SolidCheckList slist;


    public SolidCheckListDialog(Context context, SolidCheckList l) {
        slist=l;

        final AlertDialog.Builder dialog = createDefaultDialog(context, l);
        dialog.setMultiChoiceItems(slist.getStringArray(), slist.getEnabledArray(), this);
        dialog.create();
        dialog.show();
    }




    @Override
    public void onClick(DialogInterface dialog, int i, boolean isChecked) {
        slist.setEnabled(i,isChecked);
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/dialog/SolidCheckListDialog.kt
package ch.bailu.aat.views.preferences.dialog

import android.content.Context
import android.content.DialogInterface
import android.content.DialogInterface.OnMultiChoiceClickListener
import ch.bailu.aat_lib.preferences.SolidCheckList

class SolidCheckListDialog(context: Context?, private val slist: SolidCheckList) : AbsSolidDialog(),
    OnMultiChoiceClickListener {
    init {
        val dialog = createDefaultDialog(context, slist)
        dialog.setMultiChoiceItems(slist.getStringArray(), slist.getEnabledArray(), this)
        dialog.create()
        dialog.show()
    }

    override fun onClick(dialog: DialogInterface, i: Int, isChecked: Boolean) {
        slist.setEnabled(i, isChecked)
    }
}
